v4/newrelic/segments.go

```diff
@@ -378,10 +378,22 @@ func (s *MessageProducerSegment) End() {
 	if s.StartTime.isEnded() {
 		return
 	}
+	s.addAttributes(s.StartTime.Span.SetAttribute)
 	s.StartTime.Span.SetName(s.name())
 	s.StartTime.end()
 }
 
+func (s *MessageProducerSegment) addAttributes(setter func(string, interface{})) {
```

Comment:
What about the name `addMessageProducerAttrs` or something like that? When I see `addAttributes()` I assume the parameters are going to be the attributes to be added, which isn't the case here. 

(This might affect other segment types too, and I may have just missed it in other PRs if so.)